urlchecker.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+# url testing
+# source urlchecker.sh
+main_dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+excluded_directory="*/node_modules/*"
+all_urls=()
+all_files=()
+match_file()
+{
+  for (( j=0; j<=${#all_urls[@]}; j++ )); do
+    if [ ${all_urls[$j]} = ${1} ]
```

Comment:
prefer to give useful names to command arguments - e.g., at the start of the function:

`local url_to_match="$1"`